### PR TITLE
chore: update application comments

### PR DESCRIPTION
**Pull request by ls0f**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: server/application/application.go [hunk 1]
```diff
@@ -143,14 +143,13 @@ func (s *Server) List(ctx context.Context, q *application.ApplicationQuery) (*ap
 			newItems = append(newItems, *a)
 		}
 	}
-	if q.Name != nil {
```
Comment (ls0f) [inline server/application/application.go:146]: there is no need to check name is empty as FilterByName has handled this

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: server/application/application.go [hunk 2]
```diff
@@ -143,14 +143,13 @@ func (s *Server) List(ctx context.Context, q *application.ApplicationQuery) (*ap
...
-			return nil, fmt.Errorf("error filtering applications by name: %w", err)
-		}
+	// Filter applications by name
+	newItems, err = argoutil.FilterByName(newItems, q.GetName())
+	if err != nil {
+		return nil, fmt.Errorf("error filtering applications by name: %w", err)
 	}
 
-	// Filter applications by name
+	// Filter applications by project
```
Comment (todaywasawesome) [inline server/application/application.go:152]:
```suggestion
	// Filter applications by projects
```

File: server/application/application.go (hunk 1 above)
Comment (pasha-codefresh) [reply server/application/application.go:146]: @crenshaw-dev inside application.pb.go it checks if nil and after FilterByName doing another check

```
func (m *ApplicationManifestQuery) GetName() string {
	if m != nil && m.Name != nil {
		return *m.Name
	}
	return ""
}
`````

```
func FilterByName(apps []argoappv1.Application, name string) ([]argoappv1.Application, error) {
	if name == "" {
		return apps, nil
	}
	
```

Comment (pasha-codefresh) [reply server/application/application.go:146]: But i actually agree with you, it is not mandatory to remove it in this PR

**Pull request closed without merging**